src/inet/UDPEndPoint.cpp

```diff
@@ -635,6 +658,13 @@ INET_ERROR UDPEndPoint::SendMsg(const IPPacketInfo * pktInfo, System::PacketBuff
 
     // Unlock LwIP stack
     UNLOCK_TCPIP_CORE();
+
+    if (msg.GetRef() > 1)
+    {
+        /* Disconnect the temporary head pbuf from the remaining pbufs in chain */
+        msgCopy->next = NULL;
```

Comment:
It seems not easy to follow that ->next = NULL is sufficient to undo the pbuf_chain call that establishes a chain. Is there a lwip call that would do this undo? `pbuf_dechain` seems to exist but did not look if it is suitable. Naming wise it seems so.

Looking at lwip, I see that `pbuf_chain` will `pbuf_ref` on the tail - so we have an extra reference on the tail that will now not get removed. Is this true or does the sending remove it?

Could you confirm that this works ok? Maybe we could add some debug-mode local checksum validation that proves that the input buffer is not touched in any way.

Reply:
Acked, we need to use pbuf_dechain to de-chain the head. The extra reference of the original pbuf is removed after the head is de-chained.